be/src/exec/pipeline/pipeline_driver_queue.cpp

```diff
@@ -13,38 +13,45 @@ void QuerySharedDriverQueue::close() {
     _cv.notify_all();
 }
 
-void QuerySharedDriverQueue::put_back(const DriverRawPtr driver) {
+void QuerySharedDriverQueue::put(const DriverRawPtr driver) {
     int level = _compute_driver_level(driver);
     driver->set_driver_queue_level(level);
     {
         std::lock_guard<std::mutex> lock(_global_mutex);
-        _queues[level].queue.emplace(driver);
+        if (driver->driver_state() == DriverState::CANCELED) {
+            _queues[level].queue.emplace_front(driver);
```

Comment:
Is it a good choice to hide this logic in a `Queue`?

Reply:
> Is it a good choice to hide this logic in a `Queue`?

I think it's ok, `DriverQueue` is only used to manage `Driver`, it makes sense to perceive the state of driver